module/VuFind/src/VuFind/View/Helper/Root/Captcha.php

```diff
@@ -0,0 +1,126 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <crhallberg@gmail.com>
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Captcha view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <crhallberg@gmail.com>
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Captcha extends AbstractHelper
+{
+    /**
+     * Captcha services
+     *
+     * @var array
+     */
+    protected $captchas = [];
+
+    /**
+     * Config
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $config;
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Config\Config $config   Config
+     * @param array                  $captchas Captchas
+     */
+    public function __construct(\Laminas\Config\Config $config,
+        array $captchas=[]
+    ) {
+        $this->config = $config;
+        $this->captchas = $captchas;
+    }
+
+    /**
+     * Return this object
+     *
+     * @return \VuFind\View\Helper\Root\Captcha
+     */
+    public function __invoke(): \VuFind\View\Helper\Root\Captcha
+    {
+        return $this;
+    }
+
+    /**
+     * Generate HTML depending on CAPTCHA type (empty if not active).
+     *
+     * @param bool $useCaptcha Boolean of active state, for compact templating
+     * @param bool $wrapHtml   Wrap in a form-group?
+     *
+     * @return string
+     */
+    public function html(bool $useCaptcha = true, bool $wrapHtml = true): string
+    {
+        if (count($this->captchas) == 0 || !$useCaptcha) {
+            return false;
+        }
+
+        $html = '';
+        foreach ($this->captchas as $captcha) {
+            $html .= $captcha->getHtml();
+        }
+        return $html;
+    }
+
+    /**
+     * Get list of URLs with JS dependancies to load for the active CAPTCHA type.
+     *
+     * @return array
+     */
+    public function js(): array
+    {
+        $jsIncludes = [];
+        foreach ($this->captchas as $captcha) {
+            $jsIncludes = array_merge($jsIncludes, $captcha->getJsIncludes());
+        }
+        return array_unique($jsIncludes);
+    }
+
+    /**
+     * Return whether Captcha is active in the config
+     *
+     * @return bool
+     */
+    protected function active(): bool
+    {
+        return count($this->captchas) > 0 && isset($config->Captcha->forms);
```

Comment:
This seems like it will always return false because `$config` should be `$this->config` in this context.